### ISSUE TITLE
[FirebaseAuth] Error 17006, The given sign-in provider is disabled

### ISSUE THREAD

**Issue by DmytroYashchenko** [firebase/firebase-ios-sdk#11945]
### Description

Hello!
Today Auth stopped working. When I try to verify phone number (by calling PhoneAuthProvider.provider().verifyPhoneNumber method) I always receive error: "The given sign-in provider is disabled for this Firebase project. Enable it in the Firebase console, under the sign-in method tab of the Auth section."
Of course phone sign-in provider is enabled it in the Firebase console, and all regions are enabled for sms verification.
Actually - everything worked like a charm and we didn't make any changes, but today it suddenly started to throw the error.
One more detail we found - for US numbers it works correctly, but for UA (Ukraine) numbers - it produces the error. We have tried to add Ukraine to the list of regions for sms verification directly - no effect.
Please help, we have tried everything... It looks like a bug((

![Screenshot 2023-10-15 at 21 59 15](https://github.com/firebase/firebase-ios-sdk/assets/49062336/583993cb-be56-4e34-b2c3-c1e29291ddd2)
![Screenshot 2023-10-15 at 21 59 07](https://github.com/firebase/firebase-ios-sdk/assets/49062336/011dcb2b-4d08-4e73-b284-d6395a3b13e1)

### Firebase SDK Version

10.16.0

### Xcode Version

15.0

### Installation Method

CocoaPods

### Firebase Product(s)

Authentication

### Targeted Platforms

iOS

### Relevant Log Output

```shell
Error Domain=FIRAuthErrorDomain Code=17006 "The given sign-in provider is disabled for this Firebase project. Enable it in the Firebase console, under the sign-in method tab of the Auth section." UserInfo={NSLocalizedDescription=The given sign-in provider is disabled for this Firebase project. Enable it in the Firebase console, under the sign-in method tab of the Auth section., FIRAuthErrorUserInfoNameKey=ERROR_OPERATION_NOT_ALLOWED}
```

### If using Swift Package Manager, the project's Package.resolved

<details>
<summary>Expand <code>Package.resolved</code> snippet</summary>
<br>

```json

Replace this line with the contents of your Package.resolved.

```

</details>

### If using CocoaPods, the project's Podfile.lock

<details>
<summary>Expand <code>Podfile.lock</code> snippet</summary>
<br>

```yml

Replace this line with the contents of your Podfile.lock!

```

</details>

**Comment by rizafran** [comment]
Thanks for reporting, @DmytroYashchenko. The error looks related to #11886 but it seems like `verifyPhoneNumber` is not included in the affected method on [email enumeration protection](https://cloud.google.com/identity-platform/docs/admin/email-enumeration-protection). I also tried to reproduce the issue, but I'm not able to get the same error. Could you provide the detailed steps or MCVE for me to reproduce the issue?

**Comment by DmytroYashchenko** [comment]
@rizafran 
Detailed step to reproduce is just to try to call verifyPhoneNumber with "+380931112233" number, for example. Any Ukrainian number (starts with +380) causes the error. The region of device is USA, locale - English, iOS 16.0, Xcode 15.0...

**Comment by kareni3** [comment]
Hi everyone!

Have the same error. 

It all started about 1-2 weeks ago. Authorization by phone number did not work for numbers starting with “+380”. 

For example, on this service https://fir-ui-demo-84a6c.firebaseapp.com I received the error:
{
    "error": {
      "code": 503,
      "message": "Error code: 39",
      "errors": [
        {
          "message": "Error code: 39",
          "domain": "global",
          "reason": "backendError"
        }
      ]
    }
}

But at the same time if you add such a number to the test set of phone numbers, then the authorization worked well. 

However, about a day ago, even this approach of using test phone numbers stopped working and now the error looks like this:
"The given sign-in provider is disabled for this Firebase project. Enable it in the Firebase console, under the sign-in method tab of the Auth section."

I understand that sending SMS may not work for some independent reason. But in this case, even test phone numbers starting with “+380” do not work at all.

**Comment by swap-app** [comment]
I have the same issue since 14 October, and the answer from Google support ☹️:

> Hello Maksym.
> 
> I'm Oscar, thanks for reaching out. I'll be assisting you.
> 
> Thank you for the report. We have verified the issue and have found that there are some key points:
> The issue is related to the region availability not if the service is enabled in the Firebase Console.
> Services are available in nearby regions, so it is not a problem with Google data centers.
> Firebase services are healthy and stable.
> The problem seems to be localized in one region and for some carriers.
> Given this information, there is little we can do to improve the availability. We have already reached out to report the error and we recommend that you do so as well in the local carriers.
> 
> Firebase is not responsible for SMS delivery. We rely on carriers to deliver SMS in the different regions where Firebase provides the Phone Authentication service.
> 
> We apologize that we do not have better news or options for you at this time. We understand that this is an important feature for your application, so it would be good to consider other options such as other auth methods or providers. We would be happy to help you implement a different Firebase auth method.
> 
> Kind regards.
> Oscar.

**Comment by IKolosynskyi** [comment]
Here is what support answered, I was referring to Vodafone as example. Few days ago only Life was working fine, right now none of the operator works
![image](https://github.com/firebase/firebase-ios-sdk/assets/11165501/baca43da-733d-4f6f-87bf-8e2dd7944487)

**Comment by swap-app** [comment]
@rizafran what about changing the error message? it's really misleading. I spent a lot of time to find issues on my side. and I'm sure not only me.

**Comment by paulb777** [comment]
I'll reopen to improve the error message from the SDK.

**Comment by DmytroYashchenko** [comment]
> @rizafran I don't understand why test numbers are not working. There is no need of SMS delivery in this case.
> <img alt="image" width="456" src="https://user-images.githubusercontent.com/3403833/275934559-64625c17-6e32-4e72-b22d-eb053619f66c.png">
 
True, same story...

**Comment by kareni3** [comment]
> @rizafran I don't understand why test numbers are not working. There is no need of SMS delivery in this case. <img alt="image" width="456" src="https://user-images.githubusercontent.com/3403833/275934559-64625c17-6e32-4e72-b22d-eb053619f66c.png">

Yes, I wrote about this earlier. The test "+380" numbers worked for a while when the real "+380" numbers did not. It seems that most likely the developers simply decided to block all numbers starting with “+380”. Please return the test numbers back to working order. It is not SMS issue.

**Comment by OlexandrStorchak** [comment]
The same issue started for two days ago, only for Ukrainian phone numbers.
Looks like Firebase don't support Ukraine numbers any more.
Its very sad, but we still waiting official answer from Firebase support

**Comment by lisovyk** [comment]
Bumping this issue - many Ukrainian services are influenced by this! The block on test numbers is not helping.

**Comment by kabrachups** [comment]
the same problem with Ukrainian numbers. To reproduce try it with the following phone number pattern: +38050 XXX-XX-XX

**Comment by alright** [comment]
I did my best to explain the issue to tech support, but it seems they just decided to drop support for Ukraine region. Here's my conversation with them.

<img width="993" alt="Inbox 2023-10-18 19-54-37" src="https://github.com/firebase/firebase-ios-sdk/assets/154148/6fe87183-15d6-41bf-8145-aab576b7a1e0">

**Comment by paulb777** [comment]
Just heard this from a product manager:

"Hi folks, we are fully unblocking all carriers in Ukraine, so all support cases with "error: 39" in Ukraine should resolve in the next 4-5 hours. If customers still experience any issues, please let me know."

**Comment by paulb777** [comment]
Closing now. Please comment or open another issue if the problem recurs.
